src/clr-angular/utils/drag-and-drop/if-dragged.ts

```diff
@@ -17,9 +17,9 @@ import {ClrDragEventListener} from "./providers/drag-event-listener";
 @Directive({selector: "[clrIfDragged]"})
 export class ClrIfDragged<T> implements OnDestroy {
     private subscriptions: Subscription[] = [];
-    constructor(private template: TemplateRef<any>, private container: ViewContainerRef,
+    constructor(private template: TemplateRef<any>, @Optional() @SkipSelf() private container: ViewContainerRef,
                 @Optional() private dragEventListener: ClrDragEventListener<T>) {
-        if (!this.dragEventListener) {
+        if (!this.dragEventListener && !this.container) {
```

Comment:
Is there a use case where only one of these conditions is true?

Reply:
This should have been `||` instead of `&&` here. If only any of them is undefined, should throw an error.